{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "from sklearn.decomposition import PCA\n",
    "from scipy.spatial.distance import mahalanobis\n",
    "from sklearn.preprocessing import StandardScaler"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "pcc1_original = pd.read_csv(\"/home/azureuser/cloudfiles/code/Users/tsuma.thomas/DSML124/outputFiles/PCC1/test/aluminium.csv\",index_col=0)\n",
    "pcc1_reconstructed = pd.read_csv(\"/home/azureuser/cloudfiles/code/Users/tsuma.thomas/DSML124/outputFiles/pcc1_reconstructed_spc/aluminium.csv\",index_col=0)\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "metadata": {},
   "outputs": [],
   "source": [
    "pcc2_original = pd.read_csv(\"/home/azureuser/cloudfiles/code/Users/tsuma.thomas/DSML124/outputFiles/PCC2/aluminium.csv\",index_col=0)\n",
    "pcc2_reconstructed = pd.read_csv(\"/home/azureuser/cloudfiles/code/Users/tsuma.thomas/DSML124/outputFiles/pcc2_reconstructed_spc/aluminium.csv\",index_col=0)\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [],
   "source": [
    "pcc1_reconstructed.columns = pcc1_original.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "metadata": {},
   "outputs": [],
   "source": [
    "pcc2_reconstructed.columns = pcc2_original.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n",
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n"
     ]
    }
   ],
   "source": [
    "mahalanobis_distance_pcc1 = []\n",
    "for i in range(len(pcc1_original)):\n",
    "    df = pd.concat([pd.DataFrame(pcc1_original.iloc[i]).T, pd.DataFrame(pcc1_reconstructed.iloc[i]).T])\n",
    "    X = np.array(df)  # Example dataset containing u and v\n",
    "    cov_matrix = np.cov(X.T)\n",
    "    inv_cov_matrix = np.linalg.inv(cov_matrix)\n",
    "    distance = mahalanobis(pd.DataFrame(pcc1_original.iloc[i]).T.values[0], pd.DataFrame(pcc1_reconstructed.iloc[i]).T.values[0], inv_cov_matrix)\n",
    "    mahalanobis_distance_pcc1.append(distance)    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(1728, 1728)\n",
      "(1728, 1728)\n",
      "(1728, 1728)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(1728, 1728)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(1728, 1728)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(1728, 1728)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(1728, 1728)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(1728, 1728)\n",
      "(1728, 1728)\n",
      "(1728, 1728)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(1728, 1728)\n",
      "(1728, 1728)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/scipy/spatial/distance.py:1064: RuntimeWarning: invalid value encountered in sqrt\n",
      "  return np.sqrt(m)\n"
     ]
    }
   ],
   "source": [
    "mahalanobis_distance_pcc2 = []\n",
    "for i in range(len(pcc2_original)):\n",
    "    df = pd.concat([pd.DataFrame(pcc2_original.iloc[i]).T, pd.DataFrame(pcc2_reconstructed.iloc[i]).T])\n",
    "    X = np.array(df)  # Example dataset containing u and v\n",
    "    cov_matrix = np.cov(X.T)\n",
    "    inv_cov_matrix = np.linalg.inv(cov_matrix)\n",
    "    print(np.shape(inv_cov_matrix))\n",
    "    distance = mahalanobis(pd.DataFrame(pcc2_original.iloc[i]).T.values[0], pd.DataFrame(pcc2_reconstructed.iloc[i]).T.values[0], inv_cov_matrix)\n",
    "    mahalanobis_distance_pcc2.append(distance)\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 94,
   "metadata": {},
   "outputs": [],
   "source": [
    "pcc1_original['mahalanobis_distance'] = mahalanobis_distance_pcc1\n",
    "pcc2_original['mahalanobis_distance'] = mahalanobis_distance_pcc2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 95,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "count    725.000000\n",
       "mean      22.013386\n",
       "std       32.065811\n",
       "min        0.158285\n",
       "25%       10.382903\n",
       "50%       15.337891\n",
       "75%       23.961095\n",
       "max      648.896311\n",
       "Name: mahalanobis_distance, dtype: float64"
      ]
     },
     "execution_count": 95,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pcc1_original['mahalanobis_distance'].describe()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 97,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "count     5.000000\n",
       "mean     16.642487\n",
       "std       9.122115\n",
       "min       4.585131\n",
       "25%       9.351746\n",
       "50%      21.632695\n",
       "75%      22.040904\n",
       "max      25.601959\n",
       "Name: mahalanobis_distance, dtype: float64"
      ]
     },
     "execution_count": 97,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Bad pipe message: %s [b':\\x8deZ\\x07\\x1a\\xe1\\x08\\x88\\xe7\\xf5bn\\x8f\\xdb\\xc3\\xe2g \\xb6\\xa9+\\xf0\\xdd\\xb5\\x16*\\xfbB\\xe0\\\\\\x82\\xe2\\xb7\\x13g;\\xf7\\xd4)P\\xe3\\xa0\\xd8#\\x98\\x08\\x94\\xa7\\x02O\\x00\\x08\\x13\\x02\\x13\\x03\\x13\\x01\\x00\\xff\\x01\\x00\\x00\\x8f\\x00\\x00\\x00\\x0e\\x00\\x0c\\x00\\x00\\t127.0.0.1\\x00\\x0b\\x00\\x04\\x03\\x00\\x01\\x02\\x00\\n\\x00\\x0c\\x00\\n\\x00\\x1d\\x00\\x17\\x00\\x1e\\x00\\x19\\x00\\x18\\x00#\\x00\\x00\\x00\\x16\\x00\\x00\\x00\\x17\\x00\\x00\\x00\\r\\x00\\x1e\\x00\\x1c\\x04\\x03\\x05\\x03\\x06\\x03\\x08\\x07\\x08\\x08\\x08\\t\\x08\\n\\x08\\x0b\\x08\\x04\\x08\\x05\\x08\\x06\\x04\\x01\\x05\\x01\\x06\\x01\\x00+']\n",
      "Bad pipe message: %s [b'\\xa6\\xf9SH|b\\xd2$\\xcat\\x89\\xb3\\xf3\\xd1\\x1b\\x00z~\\x00\\x00\\xa6\\xc0,\\xc00\\x00\\xa3\\x00\\x9f\\xcc\\xa9\\xcc\\xa8\\xcc\\xaa\\xc0\\xaf\\xc0\\xad\\xc0\\xa3']\n",
      "Bad pipe message: %s [b\"\\xc0]\\xc0a\\xc0W\\xc0S\\xc0+\\xc0/\\x00\\xa2\\x00\\x9e\\xc0\\xae\\xc0\\xac\\xc0\\xa2\\xc0\\x9e\\xc0\\\\\\xc0`\\xc0V\\xc0R\\xc0$\\xc0(\\x00k\\x00j\\xc0s\\xc0w\\x00\\xc4\\x00\\xc3\\xc0#\\xc0'\\x00g\\x00@\\xc0r\\xc0v\\x00\\xbe\\x00\\xbd\\xc0\\n\\xc0\\x14\\x009\\x008\\x00\\x88\\x00\\x87\\xc0\\t\\xc0\\x13\\x003\\x002\\x00\\x9a\\x00\\x99\\x00E\\x00D\\xc0\\x07\\xc0\\x11\\xc0\\x08\\xc0\\x12\\x00\\x16\\x00\\x13\\x00\\x9d\\xc0\\xa1\\xc0\\x9d\\xc0Q\\x00\\x9c\\xc0\\xa0\\xc0\\x9c\\xc0P\\x00=\\x00\\xc0\\x00<\\x00\\xba\\x005\\x00\\x84\\x00/\\x00\\x96\\x00A\\x00\\x05\\x00\\n\\x00\\xff\\x01\\x00\\x00j\\x00\\x00\\x00\\x0e\\x00\\x0c\\x00\\x00\\t127.0.0.1\\x00\\x0b\\x00\\x04\\x03\\x00\\x01\\x02\\x00\\n\\x00\\x0c\\x00\\n\\x00\\x1d\\x00\\x17\\x00\\x1e\\x00\\x19\\x00\\x18\\x00\", b'\\x00\\x00\\x16\\x00\\x00\\x00\\x17\\x00\\x00\\x00\\r\\x000\\x00.\\x04\\x03\\x05\\x03\\x06\\x03\\x08\\x07\\x08\\x08\\x08\\t\\x08\\n\\x08\\x0b\\x08\\x04\\x08']\n",
      "Bad pipe message: %s [b'\\x06\\x04\\x01\\x05']\n",
      "Bad pipe message: %s [b'']\n",
      "Bad pipe message: %s [b'', b'\\x03\\x03']\n",
      "Bad pipe message: %s [b'']\n",
      "Bad pipe message: %s [b'\\x8c\\x12\\x8f\\x0f\\x0bi=\\xab&\\xda\\x8d\\x1fD\\rZ\\xf0\\xd9\\xcb\\x00\\x00>\\xc0\\x14\\xc0\\n\\x009\\x008\\x007\\x006\\xc0\\x0f\\xc0\\x05\\x005\\xc0\\x13\\xc0\\t\\x003\\x002\\x001\\x000\\xc0\\x0e\\xc0\\x04\\x00/\\x00\\x9a\\x00\\x99\\x00\\x98\\x00\\x97\\x00\\x96\\x00\\x07\\xc0\\x11\\xc0\\x07\\xc0\\x0c\\xc0\\x02\\x00\\x05\\x00\\x04\\x00\\xff\\x02\\x01\\x00\\x00C\\x00\\x00\\x00\\x0e\\x00\\x0c']\n",
      "Bad pipe message: %s [b'', b'\\x02']\n",
      "Bad pipe message: %s [b'\\x05\\x02\\x06']\n",
      "Bad pipe message: %s [b'\\x96\\xf3C\\x02Mc\\x9d\\xda\\x01\\xd4\\x10$\\xf4\\xb3\\x14 \\xe7\\xe2\\x00\\x00\\xa2\\xc0\\x14\\xc0\\n\\x009\\x008\\x007\\x006\\x00\\x88\\x00\\x87\\x00\\x86\\x00\\x85\\xc0\\x19\\x00:\\x00\\x89\\xc0\\x0f\\xc0\\x05\\x005\\x00\\x84\\xc0\\x13\\xc0\\t\\x003\\x002\\x001\\x000\\x00\\x9a\\x00\\x99\\x00\\x98\\x00\\x97\\x00E\\x00D\\x00C\\x00B\\xc0\\x18\\x004\\x00\\x9b\\x00F\\xc0\\x0e\\xc0\\x04\\x00']\n",
      "Bad pipe message: %s [b'\\x96\\x00A\\x00\\x07\\xc0\\x11\\xc0\\x07\\xc0\\x16\\x00\\x18\\xc0\\x0c\\xc0\\x02\\x00\\x05\\x00\\x04\\xc0\\x12\\xc0\\x08\\x00\\x16\\x00\\x13\\x00\\x10\\x00\\r\\xc0\\x17\\x00\\x1b\\xc0\\r\\xc0\\x03\\x00\\n\\x00\\x15\\x00']\n",
      "Bad pipe message: %s [b'\\x0f\\x00\\x0c\\x00\\x1a\\x00\\t\\x00\\x14\\x00\\x11\\x00\\x19\\x00\\x08\\x00\\x06']\n",
      "Bad pipe message: %s [b'\\xb1c\\xc3\\x10\\xb92\\xe1`\\xc7\\xf5\\x89\\x8e?F\\x802\\xdf?\\x00\\x00\\xa2\\xc0\\x14\\xc0\\n\\x009\\x008\\x007\\x006\\x00\\x88\\x00\\x87\\x00\\x86\\x00\\x85\\xc0\\x19\\x00:\\x00\\x89\\xc0\\x0f\\xc0\\x05\\x005\\x00\\x84\\xc0\\x13\\xc0\\t\\x003\\x002\\x001\\x000\\x00\\x9a\\x00\\x99\\x00\\x98\\x00\\x97\\x00E\\x00D\\x00C\\x00B\\xc0\\x18\\x004\\x00\\x9b\\x00F\\xc0\\x0e\\xc0\\x04\\x00/\\x00\\x96\\x00A\\x00\\x07\\xc0\\x11\\xc0\\x07\\xc0\\x16\\x00\\x18\\xc0\\x0c\\xc0\\x02\\x00\\x05\\x00\\x04\\xc0\\x12\\xc0\\x08\\x00\\x16\\x00\\x13\\x00\\x10\\x00\\r\\xc0\\x17\\x00\\x1b\\xc0\\r\\xc0\\x03\\x00\\n\\x00\\x15\\x00\\x12\\x00\\x0f\\x00\\x0c\\x00\\x1a\\x00\\t\\x00\\x14\\x00\\x11\\x00\\x19\\x00\\x08\\x00\\x06\\x00\\x17\\x00\\x03\\xc0\\x10\\xc0\\x06\\xc0\\x15\\xc0\\x0b\\xc0\\x01\\x00\\x02\\x00\\x01\\x00\\xff\\x02\\x01\\x00\\x00C\\x00\\x00\\x00\\x0e\\x00\\x0c\\x00\\x00\\t127.', b'0.1\\x00\\x0b\\x00\\x04\\x03\\x00\\x01\\x02\\x00\\n\\x00\\x1c\\x00\\x1a\\x00\\x17\\x00\\x19\\x00\\x1c\\x00\\x1b\\x00\\x18\\x00\\x1a\\x00\\x16\\x00\\x0e\\x00\\r\\x00\\x0b\\x00\\x0c\\x00\\t\\x00\\n\\x00#\\x00\\x00']\n",
      "Bad pipe message: %s [b'\\x92\\xe5\\xc5\\\\%\\\\y^^\\xab\\xf9\\x99,\\xa8\\xe1XX\\x86\\x00\\x00>\\xc0\\x14\\xc0\\n\\x009\\x008\\x007\\x00', b'\\x0f\\xc0\\x05\\x005\\xc0\\x13\\xc0\\t\\x003\\x002\\x001\\x000\\xc0\\x0e\\xc0\\x04\\x00/\\x00\\x9a\\x00\\x99\\x00\\x98\\x00\\x97\\x00\\x96\\x00\\x07\\xc0\\x11\\xc0\\x07\\xc0\\x0c\\xc0\\x02\\x00\\x05\\x00\\x04\\x00\\xff\\x02\\x01\\x00\\x15']\n",
      "Bad pipe message: %s [b'\\x00\\x02']\n",
      "Bad pipe message: %s [b\"\\x85{xH\\x94<\\x01eBI\\xfb\\xa6\\x15\\x9b\\x80\\x89\\x87\\x95\\x00\\x00\\x86\\xc00\\xc0,\\xc0(\\xc0$\\xc0\\x14\\xc0\\n\\x00\\xa5\\x00\\xa3\\x00\\xa1\\x00\\x9f\\x00k\\x00j\\x00i\\x00h\\x009\\x008\\x007\\x006\\xc02\\xc0.\\xc0*\\xc0&\\xc0\\x0f\\xc0\\x05\\x00\\x9d\\x00=\\x005\\xc0/\\xc0+\\xc0'\\xc0#\\xc0\\x13\\xc0\\t\\x00\\xa4\\x00\\xa2\\x00\\xa0\\x00\\x9e\\x00g\\x00@\\x00?\\x00>\\x003\\x002\\x00\", b'0\\xc01\\xc0-\\xc0)\\xc0%\\xc0\\x0e\\xc0\\x04\\x00\\x9c\\x00<\\x00/\\x00\\x9a\\x00\\x99\\x00\\x98\\x00\\x97\\x00\\x96\\x00\\x07\\xc0\\x11\\xc0\\x07\\xc0\\x0c\\xc0\\x02\\x00\\x05\\x00\\x04\\x00\\xff\\x02\\x01\\x00']\n",
      "Bad pipe message: %s [b'\\x7f\\x81\\xc0\\xf8\\x0f\\xd6\\xd4\\r\\x0b\\xf9iX,i\\xfb(\\xa7\\xea\\x00\\x00\\xf4\\xc00\\xc0,\\xc0(']\n",
      "Bad pipe message: %s [b\"\\xc0\\x14\\xc0\\n\\x00\\xa5\\x00\\xa3\\x00\\xa1\\x00\\x9f\\x00k\\x00j\\x00i\\x00h\\x009\\x008\\x007\\x006\\x00\\x88\\x00\\x87\\x00\\x86\\x00\\x85\\xc0\\x19\\x00\\xa7\\x00m\\x00:\\x00\\x89\\xc02\\xc0.\\xc0*\\xc0&\\xc0\\x0f\\xc0\\x05\\x00\\x9d\\x00=\\x005\\x00\\x84\\xc0/\\xc0+\\xc0'\\xc0#\\xc0\\x13\\xc0\\t\\x00\\xa4\\x00\\xa2\\x00\\xa0\\x00\\x9e\\x00g\\x00@\\x00?\\x00>\\x003\\x002\\x001\\x000\\x00\\x9a\\x00\\x99\\x00\\x98\\x00\\x97\\x00E\\x00D\\x00C\\x00B\\xc0\\x18\\x00\\xa6\\x00l\\x004\\x00\\x9b\\x00F\\xc01\\xc0-\\xc0)\\xc0%\\xc0\\x0e\\xc0\\x04\\x00\\x9c\\x00<\\x00/\\x00\\x96\\x00A\\x00\\x07\\xc0\\x11\\xc0\\x07\\xc0\\x16\\x00\\x18\\xc0\\x0c\\xc0\\x02\\x00\\x05\\x00\\x04\\xc0\\x12\\xc0\\x08\\x00\\x16\\x00\\x13\\x00\\x10\\x00\\r\\xc0\\x17\\x00\\x1b\\xc0\\r\\xc0\\x03\\x00\"]\n",
      "Bad pipe message: %s [b'\\x15\\x00\\x12\\x00\\x0f\\x00\\x0c\\x00\\x1a']\n"
     ]
    }
   ],
   "source": [
    "pcc2_original['mahalanobis_distance'].describe()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 91,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0.2565556343019135"
      ]
     },
     "execution_count": 91,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 90,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0.16666666666666666"
      ]
     },
     "execution_count": 90,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 92,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1411"
      ]
     },
     "execution_count": 92,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3.10 - SDK v2",
   "language": "python",
   "name": "python310-sdkv2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
